File: programmers/코딩테스트 입문 with Java/문자열계산하기.java
package 코딩테스트입문;
/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120902
 */
public class 문자열계산하기 {
    public static void main(String[] args) {
        System.out.println(solution("3 + 4"));
        System.out.println(solution("3 + 4 + 10"));
        System.out.println(solution("3 + 4 - 10"));
    }

    public static int solution(String my_string) {
        String[] arr = my_string.split(" ");
        int answer = Integer.parseInt(arr[0]);

        StringBuilder sb = new StringBuilder();
        for (int i = 1; i < arr.length - 1; i += 2) {
            switch (arr[i]) {
                case "+":
                    answer += Integer.parseInt(arr[i + 1]);
                    break;
                case "-":
                    answer -= Integer.parseInt(arr[i + 1]);
            }
        }
        return answer;
    }
}

File: programmers/코딩테스트 입문 with Python/문자열 계산하기.py
def solution(my_string):
    my_list = my_string.split(' ')
    answer = int(my_list[0])

    for i in range(2, len(my_list), 2):
        if my_list[i - 1] == '-':
            answer -= int(my_list[i])
        else:
            answer += int(my_list[i])
    return answer


print(solution("3 + 4"))
print(solution("3 + 4 + 4 + 3"))
print(solution("100000 + -100000"))
